calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java

```diff
@@ -333,23 +314,15 @@ public String eventSubscriptionReference(String context, String calendarId, Stri
 
 	} // eventSubscriptionReference
 
-	/**
-	 * Takes several calendar References and merges their events from within a given time range.
-	 * 
-	 * @param references
-	 *        The List of calendar References.
-	 * @param range
-	 *        The time period to use to select events.
-	 * @param reverseOrder
-	 * 		  CalendarEventVector object will be ordered reverse.       
-	 * @return CalendarEventVector object with the union of all events from the list of calendars in the given time range.
-	 */
+	@Override
 	public CalendarEventVector getEvents(List references, TimeRange range, boolean reverseOrder) {
 
 		CalendarEventVector calendarEventVector = null;
 
-		if (references != null && range != null)
-		{
+		if (references != null) {
+			if (range == null) {
+				range = m_timeService.newTimeRange(Instant.EPOCH, Instant.MAX);
```

Comment:
@ern omg

Reply:
Like, everything. If you don't supply a range, you get everything.

Reply:
Is it causing an issue?

Reply:
PR incoming ... the problem is that this isn't rows in a database, it's generating them (e.g., weekly meetings)